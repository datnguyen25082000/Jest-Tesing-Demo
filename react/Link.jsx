import React from "react";

const Link = (props) => {
  const onMouseEnterHandler = () => {};
  const onMouseLeaveHandler = () => {};
  return (
    <a
      href={props.href}
      onMouseEnter={onMouseEnterHandler}
      onMouseLeave={onMouseLeaveHandler}
      >{props.label}</a>
  );
};

export default Link
