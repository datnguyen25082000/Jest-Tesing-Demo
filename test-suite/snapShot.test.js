import React from "react";
import renderer from "react-test-renderer";
import Link from "../react/Link";

describe("snapshot test", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Link href="http://www.facebook.com" label="Facebook"></Link>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("property matchers", () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      createdAt: new Date(),
      name: "test user",
    };

    expect(user).toMatchSnapshot({
      createdAt: expect.any(Date),
      id: expect.any(Number)
    });
  })
});
