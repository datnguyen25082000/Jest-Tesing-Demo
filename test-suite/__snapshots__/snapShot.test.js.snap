// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test property matchers 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "test user",
}
`;

exports[`snapshot test renders correctly 1`] = `
<a
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
